### ISSUE TITLE
Projects created by Alire cannot be loaded by GPS

### ISSUE THREAD

**Issue by stcarrez** [alire-project/alire#379]
Let's create a project with Alire, add some crate dependencies, and now you want to run GPS.

GPS is unable to find the crate GNAT projects, and it has no way to find them.

I see two solutions:

1. Restore the generated GNAT project (alire/alr_build.gpr)
2. Provide a gps command in alr that launches GPS with the environment variables.

Or, hack GPS but this is probably out of scope....

I'm also wondering how Emacs is supposed to work in that environment.

**Comment by setton** [comment]
We'll definitely provide support in GPS :)

**Comment by mosteo** [comment]
I think currently the idea is to source the `alr setenv` output prior to launching GPS.

Issue #54 is about option 2. Privately I have received correspondence where people suggests `alr edit` to make it less ad-hoc.

I was informed that aggregate project files have limitations, so it was better not to rely on those going forward.

**Comment by stcarrez** [comment]
Thanks, the `alr setenv` could do the trick.

Beware that it generates an output that cannot be sourced as it prints messages like:

```
Looking for external crate: awa
Looking for external crate: ado
Looking for external crate: elada
Looking for external crate: keystoreada
Looking for external crate: security
Looking for external crate: serverfaces
Looking for external crate: servletada
Looking for external crate: utilada
Looking for external crate: utilada_xml
Looking for external crate: wikiada
Looking for external crate: xmlada
export GPR_PROJECT_PATH="..."
...
```

**Comment by mosteo** [comment]
Yup, I'm precisely fixing that unwanted output right now

**Comment by Fabien-Chouteau** [comment]
>  Privately I have received correspondence where people suggests `alr edit` to make it less ad-hoc.

I am in favor of this, especially because on Windows, `alr setenv` might not be easy to use.
User could have the option so set a preferred editor, and we can also install GNATstudio automatically when possible.


> I was informed that aggregate project files have limitations, so it was better not to rely on those going forward.

They won't work for cross projects for instance, as you cannot specify the target and run-time in an aggregate project.

**Comment by mosteo** [comment]
Implemented by #545 and #611
